app/controller/panel/TimeSlider.js

```diff
@@ -6,59 +6,101 @@ Ext.define('CpsiMapview.controller.panel.TimeSlider', {
+        slider.setValue(values);
 
-        viewModel.setData({
-            timeRangeObj: timeRangeObj
-        });
+        // hide thumb if range is noit checked
+        if (!isRange) {
+            slider.thumbs[1].disable();
+        }
+    },
+
+    /**
+     * Function to set the time to time dependent layers after slider initialization
+     * @param {Ext.slider.Multi} slider The time slider component
+     */
+    setTimeOnLayers: function (slider) {
+        var me = this;
+        var isRange = me.getViewModel().get('isRange');
+        this.onTimeChanged(slider, isRange);
+    },
+
+    /**
+     * Time slider changeComplete handler
+     * @param {Ext.slider.Multi} slider The time slider component
+     */
+    onChangeComplete: function (slider) {
+        var me = this;
+        var isRange = me.getViewModel().get('isRange');
+        me.onTimeChanged(slider, isRange);
     },
 
     /**
      * Function called in time slider changes
-     * @param {Ext.slider.Single | Ext.slider.Multi} slider The slider
-     * @param {Number} value The new value of the slider
+     * @param {Ext.slider.Multi} slider The slider
      */
-    onTimeChanged: function (slider, value) {
-        // TODO: WFS layers must be filtered
+    onTimeChanged: function (slider, isRange) {
         var me = this;
-        var layerNames = me.getView().layerNames;
-        Ext.each(layerNames, function (layerName) {
-            var layer = BasiGX.util.Layer.getLayerByName(layerName);
+        var values = slider.getValues();
+        var timeLayers = CpsiMapview.util.Layer.getTimeDependentLayers();
+        // TODO: WFS layers must be filtered
+        Ext.each(timeLayers, function (layer) {
             if (layer && (layer.getSource() instanceof ol.source.TileWMS || layer.getSource() instanceof ol.source.ImageWMS)) {
                 var wmsLayerSource = layer.getSource();
-                var sliderDate = me.getDateForSliderValue(value);
-                if (wmsLayerSource && sliderDate) {
+                if (wmsLayerSource) {
                     wmsLayerSource.updateParams({
-                        TIME: Ext.Date.format(sliderDate, 'c')
+                        TIME: me.formatWmsDateString(values, isRange)
                     });
                 }
             }
         });
     },
 
+    /**
+     * Function formatting the values array of multi-slider to a valid time
+     * parameter value
+     *
+     * @param {Number[]} values The slider values
+     * @param {Boolean} isRange Is range query or not
+     *
+     * @returns {String} The formatted time string
+     */
+    formatWmsDateString: function (values, isRange) {
+        var me = this;
+        // we must check for min / max here since the thumbs are not contrained
+        var startDateStr = me.getDateStringForSliderValue(Ext.Array.min(values));
+        var endDateStr = me.getDateStringForSliderValue(Ext.Array.max(values));
+        // TODO: format can be layer dependent....
```

Comment:
We can read this out from config and store in the layer. See the line 74 and those coming after that in the Factory for layers.

Reply:
Alright, I'll create an issue for that